qir/qat/Apps/Qat/Qat.cpp

```diff
@@ -281,20 +281,29 @@ int main(int argc, char** argv)
 
         auto profile = generator->newProfile(config.profile(), optimization_level, config.isDebugMode());
 
-        if (config.shouldGenerate())
+        if (ret == 0 && config.shouldGenerate())
```

Comment:
I have an impression that `ret` is always `0` here.

Reply:
That is true - corrected.